pkg/events/parse_args.go

```diff
@@ -314,17 +323,232 @@ func (arg CustomFunctionArgument) Value() uint64 {
 	return arg.val
 }
 
-func parseBpfAttachHelperUsage(helperArg int32) (string, error) {
-	switch helperArg {
-	case 0:
-		return "false", nil
-	case 1:
-		return "true", nil
-	case 2:
-		return "unknown", nil
-	default:
-		return "", errfmt.Errorf("unknown value got from bpf_attach event for helper usage arg")
+func parseBpfAttachHelperUsage(helperList []uint64) ([]string, error) {
+
+	var usedHelpers []string
+
+	for idx, helperName := range BpfHelpers {
+		if (helperList[idx/64] & (1 << (idx % 64))) > 0 {
+			usedHelpers = append(usedHelpers, helperName)
+		}
 	}
+
+	return usedHelpers, nil
+}
+
+var BpfHelpers = []string{
```

Comment:
We currently have 2 types in libbpfgo describing:

`type BPFProgType`
`type MapType`

With all descriptions about existing program and map types. I wonder if we should have this list there, in another type, as helper types. Or else we will start to split definitions related to libbpf internals. WDYT @yanivagman ?

Reply:
libbpf has func_id enum (https://github.com/libbpf/libbpf/blob/b5e9722ec268702b2ec62863222277148e4fe656/include/uapi/linux/bpf.h#L5751), so libbpfgo better have it too.

This string array, however, is a kind of "toString()" function, which we will need to implement in addition to the enum itself. So I'm also in favor of having it in libbpf, but it's ok if we will have it here for now until we a PR is made in libbpfgo, then we can update here to use the libbpfgo version.

@NDStrahilevitz it is unclear if you meant here or there:

> We should probably have it **there** (or at least some form of enums **here**) also as part of the effort to not require `parse-arguments`.

Reply:
i've created a PR in libbpfgo to parse BPF helpers:
https://github.com/aquasecurity/libbpfgo/pull/302